Лабораторная работа №13 по курсу Linux Pro.

Задание запустить nginx на нестандартном порту 3-мя разными способами:
  1) переключатели setsebool;
  2) добавление нестандартного порта в имеющийся тип;
  3) формирование и установка модуля SELinux.

  4) Обеспечить работоспособность приложения при включенном selinux.
	а) развернуть приложенный стенд https://github.com/mbfx/otus-linux-adm/tree/master/selinux_dns_problems; 
	б) выяснить причину неработоспособности механизма обновления зоны (см. README);
	в) предложить решение (или решения) для данной проблемы;
	г) выбрать одно из решений для реализации, предварительно обосновав выбор;
	д) реализовать выбранное решение и продемонстрировать его работоспособность


Решение:
1) Развернута виртуальная машина с помощью Vagrant. Установлен nginx и policycoreutils-python.
2) nginx сконфигурирован для работы на порту 4881
3) С помощью команды cat /var/log/audit/audit.log | grep nginx | grep 4881 был получен кусок лога с ошибкой запуска nginx
4) Затем он был передан для анализа программе audit2why: grep 1726304833.398:807 /var/log/audit/audit.log | audit2why
5) Результатом стала рекомендация по переключению параметра nis_enabled: setsebool -P nis_enabled 1
6) После переключения параметра nginx запустился на порту 4881 (см Запуск nginx 1 способ)
7) Сбросил параметр обратно
8) Добавил порт 4881 в тип http_port_t: semanage port -a -t http_port_t -p tcp 4881
9) nginx запустился (см Запуск nginx 2 способ)
10) Убрал порт 4881 из типа http_port_t
11) С помощью утилиты audit2allow был создан модуль nginx.pp: grep nginx /var/log/audit/audit.log | audit2allow -M nginx
12) Модуль был установлен в систему: semodule -i nginx.pp (см Запуск nginx 3 способ)
13) nginx запустился


Обеспечить работоспособность приложения при включенном selinux:
1) Скачал репозиторий, развернул виртуальные машины
2) Провел восстановление работоспособности согласно лабораторной
3) Альтернативный вариант восстановления работоспособности:
  3.1) Сделал запрос с клиента - получил ошибку (см скриншот 1)
  3.2) Ошибку из логов обработал утилитой audit2why, получил ошибку несовпадения контекстов безопасности
  3.3) Написал политику для директории /etc/named/: semanage fcontext -a -t named_zone_t "/etc/named(/.*)?"
  3.4) Сделал восстановление контекста безопасности для директории /etc/named/: restorecon -v -R /etc/named (см скриншот 2)
  3.5) Повторил запрос с клиента, снова получил ошибку
  3.6) Повторил чтение логов безопасности утилитой audit2why, была получена рекомендация включить флаг: setsebool -P named_write_master_zones 1 (см скриншот 3)
  3.7) Включил флаг named_write_master_zones
  3.8) Повторил запрос с клиента, запрос прошел (см скриншот 4)